test/ruleinvalidation.jl

```diff
@@ -0,0 +1,27 @@
+module RuleInvalidation
+
+using Test
+using Enzyme
+import .EnzymeRules: forward
+
+issue696(x) = x^2
+
+@test autodiff(Forward, issue696, Duplicated(1.0, 1.0))[1] ≈ 2.0
```

Comment:
We should also add `f(x) = issue696(x)` and see that `f` is invalidated.

Reply:
Added.